Add Burp Suite Pro to software installed by this Ansible role

jsf9k opened the pull request:
## 🗣 Description ##

This pull requests adds Burp Suite Pro to the software installed by this Ansible role.

## 💭 Motivation and Context ##

Resolves #37.

## 🧪 Testing ##

All pre-commit hooks and molecule tests pass.  I also built a staging AMI with these changes and verified that it indeed has Burp Suite Pro installed.  I was able to use Burp Suite Pro after licensing it.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] Tests have been added and/or modified to cover the changes in this PR.
* [x] All new and existing tests pass.

jsf9k committed:
Add TF code to allow read access to the Burp Suite files in S3

These files are the Linux installer and the license file.

jsf9k committed:
Add Ansible code to install and license Burp Suite Pro

jsf9k committed:
Add a molecule test to verify that Burp Suite Pro was installed

jsf9k committed:
Tweak a task name

jsf9k committed:
Add a variable for the Burp Suite installation prerequisites

jsf9k committed:
Allow the installer application to return 1

For some reason the installer is exiting with return code 1, even
though it appears to have successfully installed Burp Suite Pro.

jsf9k committed:
Allow a return code of 129 from the licensing process

I am seeing cases where the licensing process returns 129, but the
application appears to have been successfully licensed.

jsf9k commented on tasks/burp_suite_pro.yml:
```diff
@@ -0,0 +1,115 @@
+      # command returns a nonzero return code even when it succeeds.
+      failed_when:
+        - license_command.rc != 0
+        - license_command.rc != 13
+        - license_command.rc != 129
```
* According to `/usr/include/asm-generic/errno-base.h` (which is included indirectly by `/usr/include/errno.h`), an error code of `13` indicates `EACCES` AKA `Access denied`.
* According to `/usr/include/asm-generic/errno.h` (which is included indirectly by `/usr/include/errno.h`), an error code of `129` indicates `EKEYREJECTED` AKA `Key was rejected by service`.  This may be because the Burp Suite servers are refusing to let us use the key a second time.

jsf9k commented on tasks/burp_suite_pro.yml:
```diff
@@ -0,0 +1,115 @@
+      # command returns a nonzero return code even when it succeeds.
+      failed_when:
+        - install_command.rc != 0
+        - install_command.rc != 1
```
According to `/usr/include/asm-generic/errno-base.h` (which is included indirectly by `/usr/include/errno.h`), an error code of `1` indicates `EPERM` AKA `Operation not permitted`.

jsf9k reviewed (commented):
I made a few comments regarding the nonzero return codes I'm seeing from the Burp Suite Pro installer and licensing processes.

If we can't reuse the same key multiple times, then as a fallback we could _install_ Burp Suite Pro on the AMI but let the assessors _license_ it their personal keys (provided by NOM).  I will need to build a new AMI with these changes in order to determine if that is necessary.

hillaryj reviewed (approved):
🌬️ <- burp? Nice work :)

jsf9k committed:
Merge branch 'develop' into improvement/add-burp-suite-pro

dav3r reviewed (approved):
Looking good! 👍

jsf9k committed:
Add the 3rd-party bucket policies as outputs

We need to reuse these policies in cisagov/kali-packer.

jsf9k committed:
Merge branch 'develop' into improvement/add-burp-suite-pro

jsf9k committed:
Allow a return code of 2 from the Burp Suite Pro installer process

jsf9k committed:
Merge remote-tracking branch 'origin/improvement/add-burp-suite-pro' into improvement/add-burp-suite-pro

jsf9k committed:
Give Burp Suite test a better name

jsf9k committed:
Remove Burp Suite Pro licensing code

I found that Burp Suite Pro is fairly finicky about using the same
license too many times in succession; therefore, it makes better sense
for the users to apply the licenses they have been assigned the first
time they start Burp Suite Pro.  This is very simple and convenient
for them to do, since the Burp Suite Pro UI will simply prompt them
for the license key.
